extension/include/osqpmodulemethods.h

```diff
@@ -0,0 +1,389 @@
+    dual_inf_cert = PyArray_EMPTY(1, md, NPY_OBJECT, 0);
+
+  } else if (workspace->info->status_val == OSQP_PRIMAL_INFEASIBLE ||
+	     workspace->info->status_val == OSQP_PRIMAL_INFEASIBLE_INACCURATE) {
+    // primal infeasible
+
+    // Primal and dual solution arrays -> None values
+    x = PyArray_EMPTY(1, nd, NPY_OBJECT, 0);
+    y = PyArray_EMPTY(1, md, NPY_OBJECT, 0);
+
+    // Primal infeasibility certificate
+    prim_inf_cert = (PyObject *)PyArrayFromCArray(workspace->delta_y, md);
+
+    // Dual infeasibility certificate -> None values
+    dual_inf_cert = PyArray_EMPTY(1, nd, NPY_OBJECT, 0);
+
+    // Set objective value to infinity
+    workspace->info->obj_val = NPY_INFINITY;
+
+  } else {
+    // dual infeasible
+
+    // Primal and dual solution arrays -> None values
+    x = PyArray_EMPTY(1, nd, NPY_OBJECT, 0);
+    y = PyArray_EMPTY(1, md, NPY_OBJECT, 0);
+
+    // Primal infeasibility certificate -> None values
+    prim_inf_cert = PyArray_EMPTY(1, md, NPY_OBJECT, 0);
+
+    // Dual infeasibility certificate
+    dual_inf_cert = (PyObject *)PyArrayFromCArray(workspace->delta_x, nd);
+
+    // Set objective value to -infinity
+    workspace->info->obj_val = -NPY_INFINITY;
+  }
+
+  /*  CREATE INFO OBJECT */
+  // Store status string
+  status = PyUnicode_FromString(workspace->info->status);
+
+  // Store obj_val
+  if (workspace->info->status_val == OSQP_NON_CVX) {	// non convex
+    obj_val = PyFloat_FromDouble(Py_NAN);
+  } else {
+    obj_val = PyFloat_FromDouble(workspace->info->obj_val);
+  }
+
+#ifdef PROFILING
+
+#ifdef DLONG
+
+#ifdef DFLOAT
+  argparse_string_info = "LOLLOfffffffLf";
+#else
+  argparse_string_info = "LOLLOdddddddLd";
+#endif
+
+#else
+
+#ifdef DFLOAT
+  argparse_string_info = "iOiiOfffffffif";
+#else
+  argparse_string_info = "iOiiOdddddddid";
+#endif
+
+#endif
+
+  info_list = Py_BuildValue(argparse_string_info,
+			    workspace->info->iter,
+			    status,
+			    workspace->info->status_val,
+			    workspace->info->status_polish,
+			    obj_val,
+			    workspace->info->pri_res,
+			    workspace->info->dua_res,
+			    workspace->info->setup_time,
+			    workspace->info->solve_time,
+			    workspace->info->update_time,
+			    workspace->info->polish_time,
+			    workspace->info->run_time,
+			    workspace->info->rho_updates,
+			    workspace->info->rho_estimate
+			    );
+#else
+
+#ifdef DLONG
+
+#ifdef DFLOAT
+  argparse_string = "LOLLOffLf";
+#else
+  argparse_string = "LOLLOddLd";
+#endif
+
+#else
+
+#ifdef DFLOAT
+  argparse_string = "iOiiOffif";
+#else
+  argparse_string = "iOiiOddid";
+#endif
+
+#endif
+
+  info_list = Py_BuildValue(argparse_string_info,
+			    workspace->info->iter,
+			    status,
+			    workspace->info->status_val,
+			    workspace->info->status_polish,
+			    obj_val,
+			    workspace->info->pri_res,
+			    workspace->info->dua_res,
+			    workspace->info->rho_updates,
+			    workspace->info->rho_estimate,
+			    );
+#endif
```

Comment:
For the sake of readability, it would be good to replace `#endif` with `#endif /* ifdef PROFILING */` and similar elsewhere.